### PR TITLE
Improved logging and startup

### PR DESCRIPTION
**Pull request by binaryn3xus**
🚀 **Enhancements and Improvements:**

1. **Added Serilog for Enhanced Logging:**
   - Introduces Serilog for more robust and flexible logging capabilities.
   - Improves visibility into application behavior and diagnostics.

2. **Startup Readability Enhancement:**
   - Refactored startup logic for better readability.
   - Note: Renamed `PlexBotMain.cs` to `Program.cs`.
      - This might could be changed back. Just needs more research.

3. **Optimized String Handling:**
   - Replaced string interpolation with `StringBuilder` for improved performance.
   - Enhances efficiency in string concatenation.

4. **Improved .ENV File Handling:**
   - Enhances configuration management by refining how environment variables are loaded and processed.
   - Changed library used for .env

5. **Enhanced Validation Checks:**
   - Strengthened input validation in specific areas of the codebase.
   - Ensures data integrity and robustness.

6. **Updated Nuget Packages
   - Upgraded several packages and a few that had vulnerabilities.

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+discordClient.Ready += DiscordHelpers.ClientReady(host.Services);
+
+// Initialize and register event handlers
+UserEvents eventHandlers = new(discordClient);
+eventHandlers.RegisterHandlers();
+
+await discordClient.LoginAsync(TokenType.Bot, discordToken);
+await discordClient.StartAsync();
+
+//TODO: Is this really needed?
```
Comment (kalebbroo) [inline Program.cs]: This was taken directly from the Discord.NET example and i never questioned it. 
https://docs.discordnet.dev/guides/getting_started/first-bot.html#creating-a-discord-client

Comment (binaryn3xus) [reply Program.cs]: Oh, sorry, ignore that. That was a note for myself haha. I wanted to test something. Ill fix that in my next PR if there is another way.

File: Program.cs
Comment (kalebbroo) [inline Program.cs:1]: Unless there is a valid reason that it has to be changed i would rather leave it with the original name.

Comment (binaryn3xus) [reply Program.cs:1]: I will look into this. I have always left it as Program.cs. Surely there is a way to change it and not have to depend on the `Main` method.

Comment (binaryn3xus) [reply Program.cs:1]: Done. Rename just worked. I thought it had to be referenced somehow. Guess the compiler is just that smart.

File: PlexBot.csproj [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
-    <PackageReference Include="dotenv.net" Version="3.2.0" />
-    <PackageReference Include="Lavalink4NET.Discord.NET" Version="4.0.20" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
-    <PackageReference Include="SixLabors.Fonts" Version="2.0.4" />
-    <PackageReference Include="SixLabors.ImageSharp" Version="3.1.5" />
-    <PackageReference Include="SixLabors.ImageSharp.Drawing" Version="2.1.4" />
-  </ItemGroup>
+	<ItemGroup>
+		<PackageReference Include="Discord.Net" Version="3.15.3" />
+		<PackageReference Include="DotNetEnv" Version="3.0.0" />
```
Comment (kalebbroo) [inline PlexBot.csproj:16]: Why change?

File: Core/PlexAPI/PlexAuth.cs [hunk 3]
```diff
@@ -1,107 +1,100 @@
...
 
-namespace PlexBot.Core.PlexAPI
+public class PlexAuth
 {
-    public class PlexAuth
+    readonly string clientIdentifierKey = Environment.GetEnvironmentVariable("CLIENT_IDENTIFIER_KEY") ?? "";
+    readonly string plexAppName = Environment.GetEnvironmentVariable("PLEX_APP_NAME") ?? "";
+    readonly HttpClient httpClient = new();
+
+    //TODO: Come back to this later and make it work for ILogger in the Microsoft.Extensions.Logging namespace
```
Comment (kalebbroo) [inline Core/PlexAPI/PlexAuth.cs:9]: This class in currently unused and will be turned into a plugin. No need to spend anymore time on this.

File: Core/Extensions/ConfigurationExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿namespace PlexBot.Core.Extensions;
+
+public static class ConfigurationExtensions
+{
+    public static string GetRequiredValue<T>(this IConfiguration configuration, string configurationValueName)
+    {
+        var value = configuration[configurationValueName];
```
Comment (kalebbroo) [inline Core/Extensions/ConfigurationExtensions.cs]: Don't use var only explicit types. There are also some in DiscordHelpers and Program.

File: PlexBot.csproj (hunk 2 above)
Comment (binaryn3xus) [reply PlexBot.csproj:16]: The previous package didn't support fluent APIs with IConfiguration. This package does. If you have a preference for the old one, we can switch back and maybe write our own extension method if needed.

File: Core/Extensions/ConfigurationExtensions.cs (hunk 4 above)
Comment (binaryn3xus) [reply Core/Extensions/ConfigurationExtensions.cs]: Okay. Let's work together on a .editorconfig that has your preferred coding styles. For example, I didn't know if you preferred tradition contructors or primary constructors (where applicable).

**Comment by binaryn3xus** [comment]
New commit has an `.editorconfig` this way we can standardize on the code style. I took all the defaults and then made it a `warning` to remove the vars and use explicit types. Then used that to fix the vars.

File: Core/Extensions/ConfigurationExtensions.cs (hunk 4 above)
Comment (kalebbroo) [reply Core/Extensions/ConfigurationExtensions.cs]: I have no idea what .editorconfig does. Im showing my lack of knowledge again lol 

I will look it up and learn about it before we make one.

**Comment by kalebbroo** [comment]
I am going to have to get used to the top level being the entry point instead of a class. Everything seems to work and i think it will help with maintaining it.

**Pull request merged**